### DISCUSSION TITLE
non CUDA devices - DirectML ?

### DISCUSSION THREAD

**Discussion by raysintoasttv** [coqui-ai/TTS#2288] [General]
Hi all
I did a search on this and only came up with one thread that only had one response that wasnt sure - BUT

Im wondering if there's any possibility of a DirectML port of this? Just for those who dont have CUDA gpus but have GPUs and hardware with ML capabilities (like the 7900xtx)

Would it be as simple as installing pytorch-directML and tensorflow-directML-plugin instead of its original?

**Comment by Milor123** [comment]
I have the same problem, Have 6750 xt i dont know how use this proyect with DirectML, please help us !!

**Comment by ZacDair** [comment]
Same concerns, any support for non CUDA GPUS? or even just switching to CPU?

**Reply by SlashHarken** [reply]
You should be able to use CPU just by installing a cpu version of torch and the TTS-cpu package

**Reply by Milor123** [reply]
> You should be able to use CPU just by installing a cpu version of torch and the TTS-cpu package

And waiting  8 hours for do a small process? Lool

**Reply by Milor123** [reply]
> Same concerns, any support for non CUDA GPUS? or even just switching to CPU?

You could try BARK, it have directml support or using ROCm in linux

**Reply by erogol** [reply]
TTS can run anywhere Pytorch can run. So it is more about PyTorch than TTS

**Reply by neonarc4** [reply]
xxx bark doesnt support amd at all its just a wrapped out of ancaoda which carry 30-40gb itself and the offical repo doesnt have amd support at all the person who made it make completely unable  try urself pip install it fail else u need anacoda junk

**Comment by astra-waves4585478** [comment]
> Hi all I did a search on this and only came up with one thread that only had one response that wasnt sure - BUT
> 
> Im wondering if there's any possibility of a DirectML port of this? Just for those who dont have CUDA gpus but have GPUs and hardware with ML capabilities (like the 7900xtx)
> 
> Would it be as simple as installing pytorch-directML and tensorflow-directML-plugin instead of its original?

i saw this if it helps 
https://www.youtube.com/shorts/eTp46G4rK98

**Comment by astra-waves4585478** [comment]
https://rocm.docs.amd.com/projects/install-on-windows/en/latest/how-to/install.html#hip-install-full

check for compatibility if so you can run cuda via HIP DSK :)
